controllers/duty.ts

```diff
@@ -0,0 +1,101 @@
+import { z } from 'zod';
+import express, { Request, Response } from 'express';
+
+import {
+  createDuty,
+  findDutyByID,
+  findDutiesByQuery,
+  deleteDuty,
+  updateDuty,
+} from '../services/duty.service';
+import validateRequest from './validation';
+import { ClientError } from '../error_handling/client_errors';
+import { DutySchema, DutyQuerySchema } from '../services/schemas/duty.zschema';
+
+const router = express.Router();
+
+router.post(
+  '/',
+  validateRequest({ body: DutySchema }),
+  async (req: Request, res: Response) => {
+    try {
+      const result = await createDuty(req.body);
+      return res.status(201).json(result);
+    } catch (e) {
+      if (e instanceof Error) {
+        return res.status(500).json(e.message);
+      }
+    }
+  },
+);
+
+router.get(
+  '/:id',
+  validateRequest({ params: z.object({ id: z.string() }) }),
```

Comment:
Create a schema. don't want this to be here. `{ params: z.object({ id: z.string() }) }`

Reply:
resolved